File: elixir_gql/lib/elixir_gql/spotify/playlist.ex
defmodule ElixirGql.Spotify.Playlist do
  use Ecto.Schema
  import Ecto.Changeset

  schema "playlists" do
    field :nome, :string
    belongs_to :usuarios, ElixirGql.Spotify.Usuario, foreign_key: :usuario_id
    many_to_many :musicas, ElixirGql.Spotify.Musica, join_through: "playlists_musicas"

    timestamps()
  end

  @doc false
  def changeset(playlist, attrs) do
    playlist
    |> cast(attrs, [:nome])
    |> validate_required([:nome])
  end
end

File: ts_grpc/src/entity/Playlist.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, ManyToMany, JoinTable, JoinColumn } from 'typeorm';
import { Usuario } from './Usuario';
import { Musica } from './Musica';

@Entity()
export class Playlist {
  @PrimaryGeneratedColumn({ type: 'int'})
  ID!: number;

  @Column({ type: 'varchar', length: 100, nullable: true })
  Nome: string | undefined;

  @ManyToOne(() => Usuario, usuario => usuario.playlists, { nullable: false })
  @JoinColumn({ name: 'Usuario_ID' })
  usuario!: Usuario;

  @ManyToMany(() => Musica)
  @JoinTable()
  musicas: Musica[] | undefined;
}
